language/diem-framework/modules/DiemSystem.move

```diff
@@ -587,10 +587,30 @@ module DiemFramework::DiemSystem {
     /// `update_config_and_reconfigure`.
 
     spec module {
-       /// The permission "{Add, Remove} Validator" is granted to DiemRoot [[H14]][PERMISSION].
-       apply Roles::AbortsIfNotDiemRoot{account: dr_account} to add_validator, remove_validator;
+        invariant [global, isolated] forall addr: address where exists<DiemConfig<DiemSystem>>(addr):
+            addr == @DiemRoot;
+
+        invariant update old(DiemConfig::spec_is_published<DiemSystem>())
+            && DiemConfig::spec_is_published<DiemSystem>()
+            && old(len(DiemConfig::get<DiemSystem>().validators)) != len(DiemConfig::get<DiemSystem>().validators)
+                ==> Roles::spec_signed_by_diem_root_role();
+
+        invariant update forall addr: address where old(DiemConfig::spec_is_published<DiemSystem>())
+            && DiemConfig::spec_is_published<DiemSystem>():
+                (exists i in 0..len(DiemConfig::get<DiemSystem>().validators) where
+                    old(DiemConfig::get<DiemSystem>().validators[i].addr == addr):
+                        old(DiemConfig::get<DiemSystem>().validators[i].config) !=
+                        DiemConfig::get<DiemSystem>().validators[i].config
+                ) ==> (exists s: signer: ValidatorConfig::get_operator(addr) == Signer::address_of(s));
```

Comment:
As we discussed, I'm not sure whether this is trivial, since (depending on how it's coded in Boogie), it may be that there exists a signer for every address.

Can you use is_signer here?